src/ginger/backendPixie.nim

```diff
@@ -0,0 +1,250 @@
+                 strokeColor = color(0.0, 0.0, 0.0),
+                 fillColor = color(0.0, 0.0, 0.0, 0.0),
+                 rotateAngle: Option[(float, Point)] = none[(float, Point)]()) =
+  # A temporary function for drawCircle where a style isn't passed a long
+  if rotateAngle.isSome:
+    let (angle, around) = rotateAngle.get
+    img.rotate(angle, around)
+
+  let
+    fillPaint = Paint(kind: SolidPaint, color: fillColor)
+    strokePaint = Paint(kind: SolidPaint, color: strokeColor)
+
+  img.pxContext.fillStyle = fillPaint
+  img.pxContext.strokeStyle = strokePaint
+  img.pxContext.lineWidth = lineWidth
+  img.pxContext.strokeCircle(circle(center.toVec2, radius))
+  img.pxContext.fillCircle(circle(center.toVec2, radius))
+  img.saveState()
+
+proc drawCircle*(img: BImage, center: Point, radius: float,
+                 style: Style,
+                 rotateAngle: Option[(float, Point)] = none[(float, Point)]()) =
+  if rotateAngle.isSome:
+    let (angle, around) = rotateAngle.get
+    img.rotate(angle, around)
+
+  img.setStyle(style)
+
+  img.pxContext.strokeCircle(circle(center.toVec2, radius))
+  img.pxContext.fillCircle(circle(center.toVec2, radius))
+  img.saveState()
+
+proc getTextExtent*(text: string, font: types.Font): TextExtent =
+  # This is for now copied from the TikZ backend until I figure out how to
+  # use `computeBounds` from pixie.fonts
+  let ptY = font.size
+  let ptX = ptY * 0.5
+  result = TextExtent(
+    x_bearing: 0.0,
+    y_bearing: 0.0,
+    width: ptX * text.len.float,
+    height: ptY,
+  )
+  result.x_advance = result.width
+  result.y_advance = result.height
+
+func getTextAligns(alignKind: TextAlignKind): HorizontalAlignment =
+  # Return Pixie alignments given a Ginger text alignment
+  result = case alignKind
+    of taLeft: LeftAlign
+    of taCenter: CenterAlign
+    of taRight: RightAlign
+
+func getSlant(fs: FontSlant): string =
+  # Returns the suffix for a given slant
+  result = case fs
+    of fsNormal: "Regular"
+    of fsItalic: "Italic"
+    of fsOblique: "Oblique"
+
+proc setFont(img: BImage, font: types.Font, path: Option[string]) =
+  # Loads an given font file or loads a "hardcoded" one and 
+  var pxFont: string
+  if path.isSome:
+    pxFont = path.get
+  else:
+    # TODO better font loading
+    # Currently a skeleton of trying to load a font family according to `types.Font`
+    # It should look into system dirs and list the files
+    let suffix = if font.bold: "Bold" else: getSlant(font.slant)
+    let fileName = &"{font.family}-{suffix}.ttf"
+    pxFont = filename
+
+  img.pxContext.font = pxFont
+
+proc drawText*(img: BImage, text: string, font: types.Font, at: Point,
+               alignKind: TextAlignKind = taLeft,
+               rotate: Option[float] = none[float](),
+               rotateInView: Option[(float, Point)] = none[(float, Point)](),
+               fontPath: Option[string] = none[string]()) =
+  let tempFontPath = some("data/Arial-Bold.ttf")
```

Comment:
Hmm, we still have to find a better solution than this. 

As it stands, this line makes the code unusable. :(

Reply:
Yea I totally get it, currently I can't think of anything good, I'm open to suggestions.

https://github.com/zetashift/fontim I did these a while ago as an experiment.

Reply:
I'm open to adding `fontnim` as a dependency, given the simplicity of the library. Do you think it'd be possible to add some kind of "give me a default sans font" kind of thing? Then I think it'd be perfect.

Reply:
As an additional note, since fontim isn't really a package or something I maintain, I think it could just be something in a utils source file/folder?

Reply:
> I don't know a default font that works across all platforms :(, after some googling I don't think there is a 100% foolproof way(except shipping a default font with ginger).

I suppose I need to figure out how Cairo handles this.

> As an additional note, since fontim isn't really a package or something I maintain, I think it could just be something in a utils source file/folder?

Oh, in that case, sure! That's even simpler of course.